Use the `All Contribuors` spec and automate it

RedGuy12 opened the pull request:
### Resolves

<!--Which issue(s) does this pull request fix or resolve?-->

Resolves #

### Changes

<!--Please describe the changes made.-->

- Bump node versions used in workflows
- Choose an explicit Ubuntu version
- Move `lighthouseResult` type from types.d.ts to inline type
- Stop Prettier from formatting the contributors file
- Add contributors workflow
- Add past contributors to docs

### Reason for changes

<!--Why did you make them?-->

### Tests

<!--Have you tested this PR? If so, how? If not, please do so before submitting the PR.-->

yes, on a private repo

it only took 71 runs lol

### Agree to our policies

<!-- By submitting this issue, you agree to follow our policies. -->

-   [x] I agree to follow the [Code of Conduct](https://github.com/onedotprojects/auth/blob/issues/.github/CODE_OF_CONDUCT.md).
-   [x] I agree to follow the [Contributing Guidelines](https://github.com/onedotprojects/auth/blob/issues/.github/CONTRIBUTING.md).

wgyt commented:
please resolve the conflicts

wgyt commented:
@RedGuy12 question, why do i not have certain items while you do? to be fair you are the main maintainer, but i wrote wgytauth (became this) and changed a lot until you sort of took over the work

RedGuy12 commented:
I just looked at the issues/prs on this repo and added the related contributions...I didn't rlly think about much else outside of this repo.

I need to install the all-contributors bot so ppl can add themselves if the worlflow doesn't work. Then you can manually add yourself



________________________________
From: William Horning ***@***.***>
Sent: Sunday, August 15, 2021 7:50:38 PM
To: onedotprojects/auth ***@***.***>
Cc: Paul Reid ***@***.***>; Mention ***@***.***>
Subject: Re: [onedotprojects/auth] Use the `All Contribuors` spec and automate it (#156)


@RedGuy12<https://github.com/RedGuy12> question, why do i not have certain items while you do? to be fair you are the main maintainer, but i wrote wgytauth (became this) and changed a lot until you sort of took over the work

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/onedotprojects/auth/pull/156#issuecomment-899141338>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOT5DEWC2CDBN3RPUF64KFTT5BOF5ANCNFSM5CDWLLOQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>.
